test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -1618,6 +1618,122 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
         }
     }
 
+    Context "Invoke-WebRequest File Resume Feature" {
+        BeforeAll {
+            $outFile = Join-Path $TestDrive "resume.txt"
+
+            # Download the entire file to reference in tests
+            $referenceFile = Join-Path $TestDrive "reference.txt"
+            $uri = Get-WebListenerUrl -Test 'Resume'
+            Invoke-WebRequest -uri $uri -OutFile $referenceFile -ErrorAction Stop
+            $referenceFileHash = Get-FileHash -Algorithm SHA256 -Path $referenceFile
+            $referenceFileSize = Get-Item $referenceFile | Select-Object -ExpandProperty Length
+        }
+
+        AfterEach {
+            Remove-Item -Force -ErrorAction 'SilentlyContinue' -Path $outFile
+        }
+
+        It "Invoke-WebRequest -Resume requires -OutFile" {
+            $uri = Get-WebListenerUrl -Test 'Resume'
+
+            { Invoke-WebRequest -Resume -Uri $uri -ErrorAction Stop } |
+                Should -Throw -ErrorId 'WebCmdletOutFileMissingException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand'
+        }
+
+        It "Invoke-WebRequest -Resume Downloads the whole file when the file does not exist" {
+            $uri = Get-WebListenerUrl -Test 'Resume'
+            $response = Invoke-WebRequest -uri $uri -OutFile $outFile -Resume -PassThru
+
+            $outFileHash = Get-FileHash -Algorithm SHA256 -Path $outFile
+            $outFileHash.Hash | Should -BeExactly $referenceFileHash.Hash
+            Get-Item $outFile | Select-Object -ExpandProperty Length | Should -Be $referenceFileSize
+            $response.Headers.'X-Has-Range'[0] | Should -BeExactly 'true'
+            $response.Headers.'X-Request-Range'[0] | Should -BeExactly 'bytes=0-'
+            $response.StatusCode | Should -BeExactly 206
+            $response.Headers.'Content-Range'[0] | Should -BeExactly "bytes 0-$($referenceFileSize-1)/$referenceFileSize"
+        }
+
+        It "Invoke-WebRequest -Resume overwrites an existing file that is larger than the remote file" {
+            # Create a file larger than the download file
+            $largerFileSize = $referenceFileSize + 20
+            1..$largerFileSize | ForEach-Object { [Byte]$_ } | Set-Content -AsByteStream $outFile
+            $largerFileSize = Get-Item $outFile | Select-Object -ExpandProperty Length
+
+            $uri = Get-WebListenerUrl -Test 'Resume'
+            $response = Invoke-WebRequest -uri $uri -OutFile $outFile -Resume -PassThru
+
+            $outFileHash = Get-FileHash -Algorithm SHA256 -Path $outFile
+            $outFileHash.Hash | Should -BeExactly $referenceFileHash.Hash
+            Get-Item $outFile | Select-Object -ExpandProperty Length | Should -Be $referenceFileSize
+            Get-Item $outFile | Select-Object -ExpandProperty Length | Should -BeLessThan $largerFileSize
+            $response.Headers.'X-Has-Range'[0] | Should -BeExactly 'false'
+            $response.StatusCode | Should -BeExactly 200
```

Comment:
Number comparison should use -Be to avoid confusion for others reading the code

Reply:
fixed